### PR TITLE
Add integration test for Split MII zips across Config Maps

### PR DESCRIPTION
**Pull request by marinakog**
Since the k8s configmap only has 1m limit,  so if all those files in the configmap added up exceeded 1m then we have a problem.   In order to test this, i 
start with a JRF domain (primordial domain is already big because of all the wsm policy ) and
add other configuration : JDBC with  a very very long string in the notes. The test checks that code splits the configmap and generates multiple introspector configmaps and all servers in domain start and run. 

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4334

### CODE REVIEW

**Review by russgold — APPROVED** [review]

**Review by anpanigr — COMMENTED** [review]
Update the PR description with latest  result after re-syncing the develop branch

**Comment by marinakog** [comment]
> Update the PR description with latest result after re-syncing the develop branch

updated and also Russ fixed all issues. https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4320/ ( intermittent failures not related to my change)

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+    logger.info("Found the default service nodePort {0}", nodePort);
+    String curlCmd1 = "curl -s -L --show-error --noproxy '*' "
+        + " http://" + K8S_NODEPORT_HOST + ":" + nodePort
+        + "/em --write-out %{http_code} -o /dev/null";
+    logger.info("Executing default nodeport curl command {0}", curlCmd1);
+    assertTrue(callWebAppAndWaitTillReady(curlCmd1, 5), "Calling web app failed");
+    logger.info("EM console is accessible thru default service");
+  }
+
+  private static Integer getDBNodePort(String namespace, String dbName) {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java:415]: Move this utility to utils/DbUtils.java,  we have the same code repeated in multiple ITClasses. 

$ grep "private static Integer getDBNodePort" *.java
ItCrossDomainTransaction.java:  private static Integer getDBNodePort(String namespace, String dbName) {
ItIstioCrossClusters.java:  private static Integer getDBNodePort(String namespace, String dbName) {
ItIstioCrossDomainTransaction.java:  private static Integer getDBNodePort(String namespace, String dbName)

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+    }
+  }
+
+  /**
+   * Save the OPSS key wallet from a running JRF domain's introspector configmap to a file.
+   * @param namespace namespace where JRF domain exists
+   * @param domainUid unique domain Uid
+   * @param walletfileSecretName name of wallet file secret
+   */
+  private void saveAndRestoreOpssWalletfileSecret(String namespace, String domainUid,
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java:292]: This method is already there in ItFmwMiiDomain.java, so this is another candidate to move into a Utility class say FmwUtils.java

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+import static oracle.weblogic.kubernetes.utils.FileUtils.replaceStringInFile;
+import static oracle.weblogic.kubernetes.utils.TestUtils.callWebAppAndWaitTillReady;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Test to creat a FMW domain in model in image with over 1Mb data
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java:73]: Typo creat -> create  1Mg- 1Mb
How do you create domain image with over over  1MB of data. Do u package a big application jar

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+   * @param adminSecretName  name of admin secret
+   * @param repoSecretName name of repository secret
+   * @param encryptionSecretName name of encryption secret
+   * @param rcuAccessSecretName name of RCU access secret
+   * @param opssWalletPasswordSecretName name of opss wallet password secret
+   * @param replicaCount count of replicas
+   * @param miiImage name of model in image
+   * @return Domain WebLogic domain
+   */
+  private Domain createDomainResource(
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java:298]: Move this method to a Utility Class as the same method with same signature is defined in ItFmwMiiDomain.java

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+                .introspectorJobActiveDeadlineSeconds(600L)));
+
+    return domain;
+  }
+
+  /**
+   * Verify Pod is ready and service exists for both admin server and managed servers.
+   * Verify EM console is accessible.
+   */
+  private void verifyDomainReady(String domainNamespace, String domainUid) {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java:354]: This method is re defined in ItFmwMiiDomain.java. Move it to a new FmwUtil class.

**Review by anpanigr — COMMENTED** [review]
Run all the parallel tests after the modification and update PR description with the result

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java (hunk 3 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java:73]: it is in description of PR, I add JDBC in the model.yaml with huge data in the Notes field

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 6]
```diff
@@ -29,12 +28,12 @@
 import oracle.weblogic.domain.DomainSpec;
 import oracle.weblogic.domain.Model;
 import oracle.weblogic.domain.ServerPod;
-import oracle.weblogic.kubernetes.actions.impl.primitive.Kubernetes;
 import oracle.weblogic.kubernetes.annotations.IntegrationTest;
 import oracle.weblogic.kubernetes.annotations.Namespaces;
 import oracle.weblogic.kubernetes.assertions.TestAssertions;
 import oracle.weblogic.kubernetes.logging.LoggingFacade;
 import oracle.weblogic.kubernetes.utils.BuildApplication;
+import oracle.weblogic.kubernetes.utils.DbUtils;
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:36]: Can you use  import static oracle.weblogic.kubernetes.utils.DbUtils.getDBNodePort;

**Review by anpanigr — COMMENTED** [review]
Modify the import statements for DbUtils.getDBNodePort

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 6 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java (hunk 5 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java (hunk 4 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java]: done

**Comment by marinakog** [comment]
newest results: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4334

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java (hunk 2 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java (hunk 1 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java [hunk 7]
```diff
@@ -0,0 +1,425 @@
...
+  private static final ConditionFactory withStandardRetryPolicy
+      = with().pollDelay(2, SECONDS)
+      .and().with().pollInterval(10, SECONDS)
+      .atMost(5, MINUTES).await();
+
+
+  /**
+   * Start DB service and create RCU schema.
+   * Assigns unique namespaces for operator and domains.
+   * Pull FMW image and Oracle DB image if running tests in Kind cluster.
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java [hunk 8]
```diff
@@ -0,0 +1,425 @@
...
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.installAndVerifyOperator;
+import static oracle.weblogic.kubernetes.utils.DbUtils.setupDBandRCUschema;
+import static oracle.weblogic.kubernetes.utils.FileUtils.replaceStringInFile;
+import static oracle.weblogic.kubernetes.utils.TestUtils.callWebAppAndWaitTillReady;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java:55]: added

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java [hunk 9]
```diff
@@ -0,0 +1,425 @@
...
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@DisplayName("Test to a create JRF model in image domain and start the domain")
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java [hunk 10]
```diff
@@ -0,0 +1,425 @@
+// Copyright (c) 2020, 2021, Oracle and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwBigCMMiiDomain.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/FmwUtils.java [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+import static oracle.weblogic.kubernetes.TestConstants.K8S_NODEPORT_HOST;
+import static oracle.weblogic.kubernetes.actions.TestActions.getServiceNodePort;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkPodReadyAndServiceExists;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.getExternalServicePodName;
+import static oracle.weblogic.kubernetes.utils.TestUtils.callWebAppAndWaitTillReady;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * The common utility class for model-in-image tests.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/FmwUtils.java:32]: Change the description to "Common utility methods for FMW Domain"

**Review by anpanigr — COMMENTED** [review]
A minor  javadoc comment on FmwUtils.java

**Review by anpanigr — APPROVED** [review]
I approve it.

**Pull request merged**